### PR TITLE
[storage] add save txn option back

### PR DESCRIPTION
**Pull request by areshand**
### Test Plan
update unit test
local test works

### CODE REVIEW

File: storage/backup/backup-cli/src/backup_types/transaction/tests.rs [hunk 1]
```diff
@@ -57,8 +57,22 @@ fn end_to_end() {
     let num_txns_to_backup = total_txns - first_ver_to_backup as usize;
     let target_version = first_ver_to_backup + total_txns as Version / 2;
     let num_txns_to_restore = (target_version - first_ver_to_backup + 1) as usize;
+    let manifest_handle_1 = rt
+        .block_on(
+            TransactionBackupController::new(
+                TransactionBackupOpt {
+                    start_version: 0,
+                    num_transactions: first_ver_to_backup as usize,
```
Comment (msmouse) [inline storage/backup/backup-cli/src/backup_types/transaction/tests.rs:65]: deserves some renaming..

**Review by msmouse — APPROVED** [review]
Were you able to recreate a "full db" of the testnet?

**Review by JoshLind — APPROVED** [review]
Thanks @areshand! Did you manage to test this on our networks?

**Comment by areshand** [comment]
still testing with testnet. currently, memory cost is very high and investigating what happened

**Comment by areshand** [comment]
Probably the reason is I use `cargo run the program`, which is a debug version. 
It takes about 3.5 hrs to recover the testnet with 440M versions.

**Pull request merged**